### PR TITLE
Fix and improve tests

### PR DESCRIPTION
**Pull request by TobiGr**
Another attempt to fix some of the failing tests.

The `testViewCount()` test is quite illogical because
- the test name does not fit to what it does,
- we cannot get the subscriber count of a channel when using the search,
- it is the only method in the file. I guess someone forgot to delete it after refactoring or whatsoever. 

I guess the test's purpose was to check the channel's stream count and therefore modified and moved it.

----
Sometimes tests are good to find bugs :) I added a null check to `LinkHandlerFactory.fromUrl(String url, String baseUrl)`.

--------------------------------------------------------------------------------------------------------------------------------

[These two tests](https://github.com/TeamNewPipe/NewPipeExtractor/blob/bdbfa268355fea58d3a11ac2fae61a066f7fb693/extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorControversialTest.java#L124) drive me crazy: 

```java
@Test
 public void testGetSubtitlesListDefault() throws IOException, ExtractionException {
    // Video (/view?v=YQHsXMglC9A) set in the setUp() method has no captions => null
    assertFalse(extractor.getSubtitlesDefault().isEmpty());
}

@Test
public void testGetSubtitlesList() throws IOException, ExtractionException {
    // Video (/view?v=YQHsXMglC9A) set in the setUp() method has no captions => null
    assertFalse(extractor.getSubtitles(MediaFormat.TTML).isEmpty());
}
```

Why is the subtitle list expected to be **NOT empty**? I think the only logical result is here to have an empty list.

### CODE REVIEW

**Comment by TobiGr** [comment]
I thought again about the subtitle tests, but did not find a reason to use `assertFalse`.

**Comment by TobiGr** [comment]
Strange. The subtitle tests failed locally when using `assertFalse`, the Travis test fail when using `assertTrue`. Why?

**Comment by Stypox** [comment]
While you are at it, could you also fix these failing tests?
```
org.schabi.newpipe.extractor.services.youtube.YoutubeStreamLinkHandlerFactoryTest > getIdfromYt FAILED
    org.schabi.newpipe.extractor.exceptions.ParsingException: Malformed url: vnd.youtube://www.youtube.com/watch?v=jZViOEv90dI
        at org.schabi.newpipe.extractor.utils.Utils.getBaseUrl(Utils.java:184)
        at org.schabi.newpipe.extractor.linkhandler.LinkHandlerFactory.fromUrl(LinkHandlerFactory.java:47)
        at org.schabi.newpipe.extractor.services.youtube.YoutubeStreamLinkHandlerFactoryTest.getIdfromYt(YoutubeStreamLinkHandlerFactoryTest.java:80)
        Caused by:
        java.net.MalformedURLException: unknown protocol: vnd.youtube
            at java.base/java.net.URL.<init>(URL.java:634)
            at java.base/java.net.URL.<init>(URL.java:523)
            at java.base/java.net.URL.<init>(URL.java:470)
            at org.schabi.newpipe.extractor.utils.Utils.stringToURL(Utils.java:145)
            at org.schabi.newpipe.extractor.utils.Utils.getBaseUrl(Utils.java:182)
            ... 2 more
```

and

```
org.schabi.newpipe.extractor.services.youtube.YoutubeChannelExtractorTest$CaptainDisillusion > testMoreRelatedItems FAILED
    java.lang.IllegalArgumentException: String input must not be null
        at org.jsoup.helper.Validate.notNull(Validate.java:26)
        at org.jsoup.parser.TreeBuilder.initialiseParse(TreeBuilder.java:26)
        at org.jsoup.parser.TreeBuilder.parse(TreeBuilder.java:42)
        at org.jsoup.parser.HtmlTreeBuilder.parse(HtmlTreeBuilder.java:52)
        at org.jsoup.parser.Parser.parse(Parser.java:89)
        at org.jsoup.Jsoup.parse(Jsoup.java:31)
        at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeChannelExtractor.getPage(YoutubeChannelExtractor.java:193)
        at org.schabi.newpipe.extractor.services.DefaultTests.defaultTestMoreItems(DefaultTests.java:55)
        at org.schabi.newpipe.extractor.services.youtube.YoutubeChannelExtractorTest$CaptainDisillusion.testMoreRelatedItems(YoutubeChannelExtractorTest.java:361)
```

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorControversialTest.java [hunk 1]
```diff
@@ -123,12 +123,12 @@ public void testGetVideoStreams() throws IOException, ExtractionException {
     @Test
     public void testGetSubtitlesListDefault() throws IOException, ExtractionException {
         // Video (/view?v=YQHsXMglC9A) set in the setUp() method has no captions => null
```
[the remark this answers is not in the document]
Comment (TobiGr) [reply extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorControversialTest.java]: Ah,. that explains a lot. `T4XJQO3qol8` is not available in Germany. For this reason I got an empty list and the test failed, but  succeeded on the travis server. I'll undo that change and update the comment.

**Comment by TobiGr** [comment]
Regarding the first one: there are two links which are only extracted from youtube websites. That's why it should be save to set the service to youtube / pass the baseUrl.

Hooray! Some other tests are failing now. But not at my local machine :tada:

**Comment by TobiGr** [comment]
Reran the tests and they finished without failing :tada:

**Pull request merged**